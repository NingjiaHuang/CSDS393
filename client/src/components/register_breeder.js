import React, { Fragment, useState } from "react";
import { Link, Redirect } from "react-router-dom";
import { toast } from "react-toastify";
import 'react-toastify/dist/ReactToastify.css'

const cors = require('cors')

const BreederRegister = ({setAuth}) => {
    const [inputs, setInputs] = useState({
        username: "",
        password: "",
        reg_email: "",
        reg_phone: "",
        cattery_name: "",
        organization: "",
        owner_name: "",
        city: ""
    }); 

    const {username, password, reg_email, reg_phone, cattery_name, organization, owner_name, city} = inputs;
    const account_type = "breeder";
    
    const onChange = (e) => {
        setInputs({...inputs, [e.target.name]:e.target.value})
    }

    const onSubmitForm = async(e) => {
        e.preventDefault();
        try{
            const body = {username, password, account_type, reg_email, reg_phone, cattery_name, organization, owner_name, city};
            
            const response = await fetch("http://localhost:4020/auth/register/breeder", 
            {
                method: "POST",
                headers: {"Content-Type": "application/json"},
                body: JSON.stringify(body)
            });
            const parseRes = await response.json();
            if(parseRes.token) {
                localStorage.setItem("token", parseRes.token);
                setAuth(true);
                toast.success("Registered successfully.")
            }else{
                setAuth(false);
                toast.error(parseRes);
            }
        }catch(err){
            console.error(err.message)
        }
    }

    return (
        <Fragment>
            <h1 className="text-center my-5">Breeder Register</h1>
            <form onSubmit={onSubmitForm}>
                <label htmlFor="name">Name: </label><input id="name" type="text" name="username"  value={username} placeholder="name" onChange={e => onChange(e)} className="form-control my-2"/>
                <label htmlFor="password">Password: </label> <input id="password"type="password" name="password" placeholder="password" className="form-control my-2" value={password} onChange={e => onChange(e)}/>
                <label htmlFor="reg_email">Email: </label><input id="reg_email" type="email" name="reg_email" placeholder="email" className="form-control my-3" value={reg_email} onChange={e => onChange(e)}/>
                <label htmlFor="reg_phone">Phone Number: </label><input id="reg_phone" type="text" name="reg_phone" placeholder="phone" className="form-control my-3" value={reg_phone} onChange={e => onChange(e)}/>
                <label htmlFor="cattery_name">Cattery Name: </label><input id="cattery_name" type="text" name="cattery_name" placeholder="cattery name" className="form-control my-2" value={cattery_name} onChange={e => onChange(e)}/>
                <label htmlFor="organization">Organization: </label><input id="organization" type="text" name="organization" placeholder="organization" className="form-control my-2" value={organization} onChange={e => onChange(e)}/>
                <label htmlFor="owner_name">Owner Name: </label><input id="owner_name" type="text" name="owner_name" placeholder="owner name" className="form-control my-2" value={owner_name} onChange={e => onChange(e)}/>
                <label htmlFor="city">City: </label><input id="city" type="text" name="city" placeholder="city" className="form-control my-2" value={city} onChange={e => onChange(e)}/>
                <button className="btn btn-success btn-block" type="submit">Submit</button>
            </form>
        </Fragment>
    )
}

export default BreederRegister;

