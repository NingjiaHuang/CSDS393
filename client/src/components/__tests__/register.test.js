import {render, screen, cleanup} from '@testing-library/react';
import register_breeder from '../register_breeder';
import register_parent from '../register_parent';


import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect';

afterEach(() =>{
    cleanup();
})


test('matches snapshot of breeder register', () =>{
    const tree = renderer.create(<register_breeder/>).toJSON();
    expect(tree).toMatchSnapshot();
})


test('matches snapshot of parent register', () =>{
    const tree = renderer.create(<register_parent/>).toJSON();
    expect(tree).toMatchSnapshot();
})

