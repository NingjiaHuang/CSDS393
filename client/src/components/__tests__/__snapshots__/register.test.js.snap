// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of breeder register 1`] = `
Array [
  <h1
    className="text-center my-5"
  >
    Breeder Register
  </h1>,
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="name"
    >
      Name: 
    </label>
    <input
      className="form-control my-2"
      id="name"
      name="username"
      onChange={[Function]}
      placeholder="name"
      type="text"
      value=""
    />
    <label
      htmlFor="password"
    >
      Password: 
    </label>
     
    <input
      className="form-control my-2"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value=""
    />
    <label
      htmlFor="reg_email"
    >
      Email: 
    </label>
    <input
      className="form-control my-3"
      id="reg_email"
      name="reg_email"
      onChange={[Function]}
      placeholder="email"
      type="email"
      value=""
    />
    <label
      htmlFor="reg_phone"
    >
      Phone Number: 
    </label>
    <input
      className="form-control my-3"
      id="reg_phone"
      name="reg_phone"
      onChange={[Function]}
      placeholder="phone"
      type="text"
      value=""
    />
    <label
      htmlFor="cattery_name"
    >
      Cattery Name: 
    </label>
    <input
      className="form-control my-2"
      id="cattery_name"
      name="cattery_name"
      onChange={[Function]}
      placeholder="cattery name"
      type="text"
      value=""
    />
    <label
      htmlFor="organization"
    >
      Organization: 
    </label>
    <input
      className="form-control my-2"
      id="organization"
      name="organization"
      onChange={[Function]}
      placeholder="organization"
      type="text"
      value=""
    />
    <label
      htmlFor="owner_name"
    >
      Owner Name: 
    </label>
    <input
      className="form-control my-2"
      id="owner_name"
      name="owner_name"
      onChange={[Function]}
      placeholder="owner name"
      type="text"
      value=""
    />
    <label
      htmlFor="city"
    >
      City: 
    </label>
    <input
      className="form-control my-2"
      id="city"
      name="city"
      onChange={[Function]}
      placeholder="city"
      type="text"
      value=""
    />
    <button
      className="btn btn-success btn-block"
    >
      Submit
    </button>
  </form>,
]
`;

exports[`matches snapshot of parent register 1`] = `
Array [
  <h1
    className="text-center my-5"
  >
    Parent Register
  </h1>,
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="name"
    >
      Name: 
    </label>
    <input
      className="form-control my-2"
      id="name"
      name="username"
      onChange={[Function]}
      placeholder="name"
      type="text"
      value=""
    />
    <label
      htmlFor="password"
    >
      Password: 
    </label>
     
    <input
      className="form-control my-2"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value=""
    />
    <label
      htmlFor="reg_email"
    >
      Email: 
    </label>
    <input
      className="form-control my-3"
      id="reg_email"
      name="reg_email"
      onChange={[Function]}
      placeholder="email"
      type="email"
      value=""
    />
    <label
      htmlFor="reg_phone"
    >
      Phone Number: 
    </label>
    <input
      className="form-control my-3"
      id="reg_phone"
      name="reg_phone"
      onChange={[Function]}
      placeholder="phone"
      type="text"
      value=""
    />
    <label
      htmlFor="preferred_name"
    >
      Preferred Name: 
    </label>
    <input
      className="form-control my-2"
      id="preferred_name"
      name="preferred_name"
      onChange={[Function]}
      placeholder="preferred name"
      type="text"
      value=""
    />
    <button
      className="btn btn-success btn-block"
    >
      Submit
    </button>
  </form>,
]
`;
